components/PostImage/PostImage.js

```diff
@@ -2,15 +2,50 @@ import styles from './PostImage.module.css';
 import cx from 'classnames';
 import React from 'react';
 
+const useElementOnScreen = (options) => {
+  const imgRef = React.useRef(null);
+  const [isVisible, setIsVisible] = React.useState(false);
+  React.useEffect(() => {
+    const observer = new IntersectionObserver(([entry]) => {
+      setIsVisible(entry.isIntersecting);
+    }, options);
+    if (imgRef.current) {
+      observer.observe(imgRef.current);
+    }
+    return () => {
+      if (imgRef.current && isVisible) {
```

Comment:
As we agreed. You don't need to check if the item is visible before you disconnect. This is a memory leak.
```suggestion
      if (imgRef.current) {
```

Reply:
Yupe.